api/client/service/opts.go

```diff
@@ -253,6 +307,21 @@ func (m *MountOpt) Value() []swarm.Mount {
 	return m.values
 }
 
+func getMountType(fields []string) (swarm.MountType, []string) {
```

Comment:
>If it helps, I can submit a PR

A second PR will not help. I'd like to get others feedback before proceeding. I think this implementation gets us closer to a user friendly mount syntax. I understand your concerns, but I don't agree. This does not suffer from the same problems as the volume syntax and forking the grammer on the type allows us to remove redundant prefixes on keys, and keeps the aliases from confusing users further. 

@thaJeztah already expressed concern in the issue that accepting some aliases for both wouldn't make much sense (ex `bind,name=/path`).